pandas/core/internals/blocks.py

```diff
@@ -2211,12 +2212,6 @@ def set(self, locs, values):
 
         self.values[locs] = values
 
-    def external_values(self):
-        return np.asarray(self.values.astype("datetime64[ns]", copy=False))
-
-    def array_values(self) -> ExtensionArray:
-        return DatetimeArray._simple_new(self.values)
```

Comment:
This is no the same as `self._holder(self.values)` as you did above, this was calling `_simple_new` on purpose for efficience

Reply:
True.  My thought was that since the DTA constructor is relatively strict, the overhead wouldn't be too bad.  Can revert if you have a strong opinion.  (and/or we could cache_readonly array_values)

Reply:
I would need to look back or time it again to be sure about the exact impact, but as far as I remember, I did this change when optimizing `extract_array`, and this was done for a reason